src/cascadia/TerminalSettingsModel/ActionArgsMagic.h

```diff
@@ -0,0 +1,160 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- ActionArgsMagic.h
+
+Abstract:
+- Contains helpers for x-macro defining all our action args. This doesn't
+  contain the actual x-macros themselves, but does contain the logic for
+  actually synthesizing an ActionArgs implementation.
+- Most of the time, you'll only need ACTION_ARGS_STRUCT
+- ACTION_ARG_BODY is for when you've got some other logic to add to the class.
+
+Author(s):
+- Mike Griese - December 2021
+
+--*/
+
+// MACRO HACKS
+//
+// We want to have code that looks like:
+//
+// FooArgs(const ParamOne& one, const ParamTwo& two) :
+//     _One{ one }, _Two{ two } {};
+//
+// However, if we just use the x-macro for this straight up, then the list will
+// have a trailing comma at the end of it, and won't compile. So, we're creating
+// this placeholder size-0 struct. It's going to be the last param for all the
+// args' ctors. It'll have a default value, so no one will need to know about
+// it. This will let us use the macro to populate the ctors as well.
+
+struct InitListPlaceholder
+{
+};
+
+//
+// The complete ActionAndArgs definition. Each macro above ACTION_ARGS_STRUCT is
+// some element of the class definition that will use the x-macro.
+//
+// You'll author a new arg by:
+//   1: define a new x-macro above with all it's properties
+//   2. Define the class with:
+//
+//   ACTION_ARGS_STRUCT(MyFooArgs, MY_FOO_ARGS);
+//
+// In that macro, we'll use the passed-in macro (MY_FOO_ARGS) with each of these
+// macros below, which will generate the various parts of the class body.
+//
+// Trying to make changes here? I'd recommend godbolt with the `-E` flag to gcc.
+// That'll output the expanded macros, so you can see how it will all get
+// expanded. Pretty critical for tracking down extraneous commas, etc.
+
+// Property definitions, and JSON keys
+#define DECLARE_ARGS(type, name, jsonKey, required, ...)    \
+    static constexpr std::string_view name##Key{ jsonKey }; \
+    ACTION_ARG(type, name, ##__VA_ARGS__);
+
+// Parameters to the non-default ctor
+#define CTOR_PARAMS(type, name, jsonKey, required, ...) \
+    const type &name##Param,

```

Comment:
Why don't you just remove the trailing comma here? Then you won't need `InitListPlaceholder` either.